component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -183,48 +215,44 @@ public void connect() throws ConnectionUnavailableException {
         if (idleTimeoutString != null) {
             configuration.setIdleTimeoutInMillis(idleTimeout);
         }
+        if (sslEnabled) {
+            configuration.setTrustStoreFile(this.tlsstruststorePath);
+            configuration.setTrustStorePass(this.tlsstruststorePass);
+        }
         WebSocketClientConnector clientConnector = httpConnectorFactory.createWsClientConnector(configuration);
-        HandshakeFuture handshakeFuture = clientConnector.connect(connectorListener);
+        ClientHandshakeFuture handshakeFuture = clientConnector.connect();
+        handshakeFuture.setWebSocketConnectorListener(connectorListener);
         WebSocketSinkHandshakeListener handshakeListener = new WebSocketSinkHandshakeListener
                 (streamDefinition, semaphore);
         try {
-            handshakeFuture.setHandshakeListener(handshakeListener);
+            handshakeFuture.setClientHandshakeListener(handshakeListener);
             semaphore.acquire();
         } catch (InterruptedException e) {
             log.error("Error occurs while connecting with the server defined in " + streamDefinition, e);
         }
-        AtomicReference<Session> sessionAtomicReference = handshakeListener.getSessionAtomicReference();
-        session = sessionAtomicReference.get();
+        AtomicReference<WebSocketConnection> sessionAtomicReference =
+                handshakeListener.getWebSocketConnectionAtomicReference();
+        webSocketConnection = sessionAtomicReference.get();
     }
 
     @Override
     public void publish(Object payload, DynamicOptions dynamicOptions) throws ConnectionUnavailableException {
-        try {
-            if (session != null) {
-                if (payload instanceof ByteBuffer) {
-                    byte[] byteMessage = ((ByteBuffer) payload).array();
-                    ByteBuffer binaryMessage = ByteBuffer.wrap(byteMessage);
-                    session.getBasicRemote().sendBinary(binaryMessage);
-                } else {
-                    session.getBasicRemote().sendText(payload.toString());
-                }
+        if (webSocketConnection != null) {
+            if (payload instanceof ByteBuffer) {
+                byte[] byteMessage = ((ByteBuffer) payload).array();
+                ByteBuffer binaryMessage = ByteBuffer.wrap(byteMessage);
+                webSocketConnection.pushBinary(binaryMessage);
```

Comment:
If we want this call to be blocking we will have to call `webSocketConnection.pushBinary(binaryMessage).sync()` or we might have to listen on the returned future. Same applies for `pushText` and `terminateConnection` functions as well in all places.

Reply:
We don't wait for an acknowledgement from the servers in Siddhi sink. It will be in fire and forget mode except for connection unavailable.